### PR TITLE
Enable Anconda for enterprise build on master

### PR DESCRIPTION
**Pull request by lij55**
This patch create a successfully installer of GPDB with bundled Anaconda. 
Installcheck and unittest are checked and successfully. 
Only RHEL is supported now.

### CODE REVIEW

File: src/test/regress/output/mapred.source [hunk 1]
```diff
@@ -96,8 +96,8 @@ return sys.version_info
 $$ language plpythonu NO SQL;
 SELECT python_version() FROM env GROUP BY python_version;
     python_version     
------------------------
- (2, 6, 2, 'final', 0)
+-----------------------------------------------------------------------------
+ sys.version_info(major=2, minor=7, micro=9, releaselevel='final', serial=0)
```
Comment (hlinnaka) [inline src/test/regress/output/mapred.source]: We probably should stop printing the python version here. I don't see what it has to do with gpmapreduce, and we cannot expect people to use a particular version of python anyway. (I've been getting a diff from this on my laptop for as long as I remember)

File: gpMgmt/bin/Makefile [hunk 2]
```diff
@@ -166,10 +166,18 @@ SETUP_TOOLS_DIR=setuptools-$(SETUP_TOOLS_VERSION)
 PARSE_DIR=parse-$(PARSE_VERSION)
 ARG_PARSE_DIR=argparse-$(ARG_PARSE_VERSION)
 PYTHONSRC_INSTALL=$(PYLIB_SRC_EXT)/install
+ifneq "$(findstring rhel,$(BLD_ARCH))" ""
```
Comment (hlinnaka) [inline gpMgmt/bin/Makefile]: What does this do? Why is the python version tied to the distribution?

File: gpMgmt/bin/gppylib/commands/test/unit/test_unit_base.py [hunk 3]
```diff
@@ -21,6 +21,7 @@ def test_print_progress(self, mock1):
         w.addCommand(c2)
         w.join()
         w.print_progress(2)
+        w.haltWork();
```
Comment (hlinnaka) [inline gpMgmt/bin/gppylib/commands/test/unit/test_unit_base.py]: What's this for?

File: src/test/regress/output/mapred.source (hunk 1 above)
Comment (lij55) [inline src/test/regress/output/mapred.source]: @ivannovick , what's the current status of gpmapreduce? Do we still maintain it? ( apologize that I don't know that ...)

Comment (danielgustafsson) [inline src/test/regress/output/mapred.source]: There was a discussion around the time of the `--enable-mapreduce` which concluded that there are customers using GPMapreduce in production so it's supported and maintained.

File: gpMgmt/bin/Makefile (hunk 2 above)
Comment (lij55) [inline gpMgmt/bin/Makefile]: Anaconda is supported on RHEL only. For other platform, we will use old python 2.6 as before.
I will update the commit message.

File: gpMgmt/bin/gppylib/commands/test/unit/test_unit_base.py (hunk 3 above)
Comment (lij55) [inline gpMgmt/bin/gppylib/commands/test/unit/test_unit_base.py]: Our implement of "Working Queue" has bug if works with python2.7. It must be termicated by haltWork explicitly.
Most of similar issues are fixed before OSS. This is the missing one.

File: src/test/regress/output/mapred.source (hunk 1 above)
Comment (lij55) [inline src/test/regress/output/mapred.source]: If we need to refine the test code of mapreduce, it is better to create a new PR for that. Besides python, it also check the version of perl.
@hlinnaka , do you think we have to fix it in this PR?

Comment (ivannovick) [inline src/test/regress/output/mapred.source]: yes gpmapreduce is still in use

Comment (hlinnaka) [inline src/test/regress/output/mapred.source]: On 23/03/16 05:08, Jasper Li wrote:
>> @@ -96,8 +96,8 @@ return sys.version_info
>>   $$ language plpythonu NO SQL;
>>   SELECT python_version() FROM env GROUP BY python_version;
>>       python_version
>> ------------------------
>> - (2, 6, 2, 'final', 0)
>> +-----------------------------------------------------------------------------
>> + sys.version_info(major=2, minor=7, micro=9, releaselevel='final', serial=0)
>
> If we need to refine the test code of mapreduce, it is better to create a new PR for that. Besides python, it also check the version of perl.
> @hlinnaka , do you think we have to fix it in this PR?

It can be a separate PR. Would then be better to do that other PR first, 
to avoid the churn here, but it's not a big deal.

- Heikki

File: gpAux/Makefile [hunk 4]
```diff
@@ -861,6 +861,9 @@ endif
 ifneq "$($(BLD_ARCH)_PYTHON)" ""
 	mkdir -p $(LOADERSINSTLOC_EXT)/python
 	(cd $($(BLD_ARCH)_PYTHON) && $(TAR) cf - $(BLD_PYTHON_FILESET)) | (cd $(LOADERSINSTLOC_EXT)/python/ && $(TAR) xpf -)$(check_pipe_for_errors)
+ifneq "$(findstring rhel,$(BLD_ARCH))" ""
+	(cd $(BLD_PYTHON) && ./init.sh )
+endif
```
Comment (yydzero) [inline gpAux/Makefile]: Where is init.sh?

Comment (lij55) [inline gpAux/Makefile]: init.sh contain the initialisation code of Anaconda.
It locates in the python tar package. For example, greenplum-db-devel/ext/python/init.sh

Comment (lij55) [inline gpAux/Makefile]: It is stored in path  http://repo.eng.pivotal.io/artifactory/simple/gpdb-ext-release-local/Conda/miniconda

content of init.sh is :
`#!/bin/bash
PREFIX="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
PYTHON="$PREFIX/pkgs/python-2.7.9-1/bin/python -E"
CONDA_INSTALL="$PREFIX/pkgs/conda-3.9.1-py27_0/lib/python2.7/site-packages/conda/install.py"
$PYTHON $CONDA_INSTALL --prefix=$PREFIX --pkgs-dir=$PREFIX/pkgs --link-all`

**Comment by lij55** [comment]
Scoped changed.
Close it without merge.
